Core/include/Acts/Definitions/PdgParticle.hpp

```diff
@@ -41,4 +45,80 @@ static constexpr PdgParticle makeAbsolutePdgParticle(PdgParticle pdg) {
   return static_cast<PdgParticle>((0 <= value) ? value : -value);
 }
 
+/// Check if the PDG belongs to a nucleus, i.e. if it has 10 digits.
+/// See PDG section "Monte Carlo Particle Numbering Scheme", point 16:
+/// https://pdg.lbl.gov/2025/reviews/rpp2024-rev-monte-carlo-numbering.pdf
+static constexpr bool isNucleus(PdgParticle pdg) {
+  const auto pdgNum = static_cast<std::int32_t>(pdg);
+  return std::abs(pdgNum) > 1e9;
+}
+
+/// Convert an excited nucleus to its ground state. PDG number of a nucleus has
+/// a form 10LZZZAAAI, where I is isomer level; I=0 is the ground state. Leave
+/// other particles as-is. See PDG section "Monte Carlo Particle Numbering
+/// Scheme", point 16:
+/// https://pdg.lbl.gov/2025/reviews/rpp2024-rev-monte-carlo-numbering.pdf
+static constexpr PdgParticle makeNucleusGroundState(PdgParticle pdg) {
+  if (isNucleus(pdg)) {
+    const auto pdgNum = static_cast<std::int32_t>(pdg);
+    // set isomer level to zero
+    return static_cast<PdgParticle>((pdgNum / 10) * 10);
+  } else {
+    return pdg;
+  }
+}
+
+/// Extract Z and A for a given nucleus. PDG number of a nucleus has a form
+/// 10LZZZAAAI, where L is number of lambdas, ZZZ is proton number, AAA is
+/// atomic number, I is isomer level See PDG section "Monte Carlo Particle
+/// Numbering Scheme" , point 16:
+/// https://pdg.lbl.gov/2025/reviews/rpp2024-rev-monte-carlo-numbering.pdf
+static constexpr std::optional<std::pair<std::int32_t, std::int32_t>>
+extractNucleusZandA(PdgParticle pdg) {
+  if (isNucleus(pdg)) {
+    const auto pdgNum = static_cast<std::int32_t>(pdg);
+    // proton number respects the charge
+    int Z = (pdgNum / 10000) % 1000;
+    // atomic number is always positive
+    int A = std::abs((pdgNum / 10) % 1000);
+    return std::make_pair(Z, A);
+  } else {
+    return std::nullopt;
+  }
+}
+
+/// Calculate the mass of a nucleus using Bethe-Weizsacker formula
+/// Parameters obtained from https://www.actaphys.uj.edu.pl/R/37/6/1833
+static constexpr std::optional<float> calculateNucleusMass(PdgParticle pdg) {
```

Comment:
I think this should be moved into an anonymous namespace into `Core/src/Definitions/ParticleData.cpp` as it seems more like a detail for particle data than PDG